src/ray/gcs/gcs_server/gcs_actor_scheduler.cc

```diff
@@ -450,6 +462,9 @@ void RayletBasedActorScheduler::HandleWorkerLeaseReply(
           << actor->GetActorID()
           << " has been already cancelled. The response will be ignored. Job id = "
           << actor->GetActorID().JobId();
+      if (actor->GetState() == rpc::ActorTableData::DEAD) {
```

Comment:
If actor is dead, why we still need to kill actor?

Reply:
The reason is that there is a race condition:

sending leasing request  -> successfully leased, response sending back (not received) -> kill actor (actor dead) -> receive the response[the place you comment].